lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp

```diff
@@ -62,6 +63,10 @@ UsdUndoInsertChildCommand::UsdUndoInsertChildCommand(const UsdSceneItem::Ptr& pa
     const auto& childPrim = child->prim();
     const auto& parentPrim = parent->prim();
 
+    // Apply restriction rules
+    ufe::applyCommandRestriction(childPrim, "reparent");
+    ufe::applyCommandRestriction(parentPrim, "reparent");
+
```

Comment:
@ppt-adsk here I am applying the restriction rules very early. These rules are applied both to the child and parent prims.

Reply:
Is this really the only change/fix here?  Other than name changes I see a couple places where we've removed nullptr checks and then this is the only real change, do I have that right?

Reply:
Hey @fowlertADSK I have broken my PR into 3 parts to make it easier to review:

1- c35855050166920ad0b2f6bdc223fa56cd135fae
2- 0202eecc5b6e41b3578d092ef468feadc45e42fb
3- baf159546a2d50b71cd9738103cd9624500d96e7

The 2 line changes that you pointed out are part of section 2. There are also 2 important changes in section 2 which are related to cleaning up _childLayer and _parentLayer.

Section 3 (Fix failing unit tests) is what affects your earlier changes. If I recall correctly you told me last time that you were planning to make changes to testGroupCmd. I wanted to make sure we are aligned on this.